File: client-sdks/ruby/lib/vertexcache/model/get_result.rb
# frozen_string_literal: true

# ------------------------------------------------------------------------------
# Copyright 2025 to Present, Jason Lam - VertexCache (https://github.com/vertexcache/vertexcache)
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# ------------------------------------------------------------------------------

require_relative 'command_result'

module VertexCache
  module Model
    class GetResult < CommandResult
      attr_reader :value

      def initialize(success, message, value)
        super(success, message)
        @value = value
      end
    end
  end
end

File: client-sdk/java/sdk/src/main/java/com/vertexcache/sdk/result/GetResult.java
package com.vertexcache.sdk.result;

public class GetResult extends CommandResult {
    private final String value;

    public GetResult(boolean success, String message, String value) {
        super(success, message);
        this.value = value;
    }

    public String getValue() {
        return value;
    }
}
